server/src/main/java/org/elasticsearch/indices/recovery/RemoteRecoveryTargetHandler.java

```diff
@@ -74,25 +91,32 @@ public RemoteRecoveryTargetHandler(long recoveryId, ShardId shardId, TransportSe
                 .withType(TransportRequestOptions.Type.RECOVERY)
                 .withTimeout(recoverySettings.internalActionTimeout())
                 .build();
-
     }
 
     @Override
     public void prepareForTranslogOperations(int totalTranslogOps, ActionListener<Void> listener) {
-        transportService.sendRequest(targetNode, PeerRecoveryTargetService.Actions.PREPARE_TRANSLOG,
-            new RecoveryPrepareForTranslogOperationsRequest(recoveryId, shardId, totalTranslogOps),
-            TransportRequestOptions.builder().withTimeout(recoverySettings.internalActionTimeout()).build(),
-            new ActionListenerResponseHandler<>(ActionListener.map(listener, r -> null),
-                in -> TransportResponse.Empty.INSTANCE, ThreadPool.Names.GENERIC));
+        final TimeValue timeout = recoverySettings.internalActionTimeout();
+        final RecoveryPrepareForTranslogOperationsRequest request =
+            new RecoveryPrepareForTranslogOperationsRequest(recoveryId, shardId, totalTranslogOps);
+        final Consumer<ActionListener<Void>> action = actionListener ->
+            transportService.sendRequest(targetNode, PeerRecoveryTargetService.Actions.PREPARE_TRANSLOG, request,
+                TransportRequestOptions.builder().withTimeout(recoverySettings.internalActionTimeout()).build(),
+                new ActionListenerResponseHandler<>(ActionListener.map(actionListener, r -> null),
+                    in -> TransportResponse.Empty.INSTANCE, ThreadPool.Names.GENERIC));
+        executeRetryableAction(action, listener, timeout);
```

Comment:
using the internalActionTimout (which is 15min) as the maximum retry timeout here is I think too much to wait for a recovery to continue. I would prefer for the recovery to be failed much earlier, probably more in the range of a minute or so. Let's add a new setting in RecoverySettings (undocumented) that allows setting this for all actions here.